luwak/src/main/java/uk/co/flax/luwak/CandidateMatcher.java

```diff
@@ -151,8 +152,9 @@ protected T matches(String docId, String queryId) {
         Map<String, DocumentMatches<T>> results = new HashMap<>();
         for (InputDocument doc : docs) {
             String id = doc.getId();
-            if (matches.containsKey(id))
-                results.put(id, new DocumentMatches<>(id, matches.get(id).matches.values()));
+            MatchHolder<T> matchHolder = matches.get(id);
+            if (matchHolder != null)
+                results.put(id, new DocumentMatches<>(id, matchHolder.matches.values()));
             else
```

Comment:
And similarly here

Reply:
In this method it's modifying the result map rather than the original map it's accessing, so I'm not sure compute makes sense in this case.

Reply:
D'oh, yes, of course.